python/pyspark/sql/dataframe.py

```diff
@@ -2911,6 +2911,37 @@ def freqItems(
             support = 0.01
         return DataFrame(self._jdf.stat().freqItems(_to_seq(self._sc, cols), support), self.sql_ctx)
 
+    def withColumns(self, colsMap: Dict[str, Column]) -> "DataFrame":
```

Comment:
I admit that this is sort of not so pretty in Python context.

Reply:
Or we could at least make this `colsMap` as keyword arguments .. ?

Reply:
We can switch to `*colsMap` now (in the future we can switch to positional only)

```python
def withColumns(self, *colsMap: Dict[str, Column]) -> "DataFrame":
    assert len(colsMap) == 1
    ...
```

to cleanly enable `**kwargs` in the future, if there is enough support for such feature.

Reply:
Yep, that means for now, we only allow:
```python
withColumns({"col1": col1, "col2": col2})
```
in future, we can also enable kwargs to allow:
```python
# With args and kwargs
withColumns({"col1": col1, "col2": col2}, col3=col3)
# With only kwargs
withColumns(col4=col4)
```

If no objection, I will update to *colsMap in this PR.